ci: Some workflows have problems during the dependency installation phase.

This may be caused by updating the [ubuntu-latest image to ubuntu 24.04](https://github.com/actions/runner-images/issues/10636).
log ref: https://github.com/uutils/findutils/actions/runs/12856099736/job/35842390618

```
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [142 B]
Hit:6 https://packages.microsoft.com/repos/azure-cli noble InRelease
Get:7 https://packages.microsoft.com/ubuntu/24.04/prod noble InRelease [3600 B]
Hit:2 http://azure.archive.ubuntu.com/ubuntu noble InRelease
Get:3 http://azure.archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Get:4 http://azure.archive.ubuntu.com/ubuntu noble-backports InRelease [126 kB]
Get:5 http://azure.archive.ubuntu.com/ubuntu noble-security InRelease [126 kB]
Get:[8](https://github.com/uutils/findutils/actions/runs/12856099736/job/35842390618#step:5:9) https://packages.microsoft.com/ubuntu/24.04/prod noble/main amd64 Packages [17.2 kB]
Get:[9](https://github.com/uutils/findutils/actions/runs/12856099736/job/35842390618#step:5:10) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 Packages [775 kB]
Get:[10](https://github.com/uutils/findutils/actions/runs/12856099736/job/35842390618#step:5:11) http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 Components [151 kB]
Get:11 http://azure.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Packages [974 kB]
Get:12 http://azure.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Components [310 kB]
Get:13 http://azure.archive.ubuntu.com/ubuntu noble-updates/restricted amd64 Components [212 B]
Get:14 http://azure.archive.ubuntu.com/ubuntu noble-updates/multiverse amd64 Components [940 B]
Get:15 http://azure.archive.ubuntu.com/ubuntu noble-backports/main amd64 Components [208 B]
Get:16 http://azure.archive.ubuntu.com/ubuntu noble-backports/universe amd64 Components [[11](https://github.com/uutils/findutils/actions/runs/12856099736/job/35842390618#step:5:12).7 kB]
Get:17 http://azure.archive.ubuntu.com/ubuntu noble-backports/restricted amd64 Components [216 B]
Get:18 http://azure.archive.ubuntu.com/ubuntu noble-backports/multiverse amd64 Components [2[12](https://github.com/uutils/findutils/actions/runs/12856099736/job/35842390618#step:5:13) B]
Get:19 http://azure.archive.ubuntu.com/ubuntu noble-security/main amd64 Components [7252 B]
Get:20 http://azure.archive.ubuntu.com/ubuntu noble-security/universe amd64 Components [52.0 kB]
Get:21 http://azure.archive.ubuntu.com/ubuntu noble-security/restricted amd64 Components [212 B]
Get:22 http://azure.archive.ubuntu.com/ubuntu noble-security/multiverse amd64 Components [212 B]
Fetched 2682 kB in 1s (4721 kB/s)
Reading package lists...
Reading package lists...
E: You must put some 'deb-src' URIs in your sources.list
Error: Process completed with exit code 100.
```

--- a/.github/workflows/compat.yml
+++ b/.github/workflows/compat.yml
@@ -102,7 +102,7 @@ jobs:
         shell: bash
         run: |
           # Enable sources & install dependencies
-          sudo find /etc/apt/sources.list* -type f -exec sed -i 'p; s/^deb /deb-src /' '{}' +
+          sudo sed -i 's/^Types: deb$/Types: deb deb-src/' /etc/apt/sources.list.d/ubuntu.sources
           sudo apt-get update
           sudo apt-get build-dep bfs
       - name: Run BFS tests

--- a/.github/workflows/compat.yml
+++ b/.github/workflows/compat.yml
@@ -52,7 +52,7 @@ jobs:
           name: gnu-result
           path: gnu-result.json
       - name: Download the result
-        uses: dawidd6/action-download-artifact@v6
+        uses: dawidd6/action-download-artifact@v7
         with:
           workflow: compat.yml
           workflow_conclusion: completed
@@ -61,7 +61,7 @@ jobs:
           branch: main
           path: dl
       - name: Download the log
-        uses: dawidd6/action-download-artifact@v6
+        uses: dawidd6/action-download-artifact@v7
         with:
           workflow: compat.yml
           workflow_conclusion: completed
@@ -121,7 +121,7 @@ jobs:
           name: bfs-result
           path: bfs-result.json
       - name: Download the result
-        uses: dawidd6/action-download-artifact@v6
+        uses: dawidd6/action-download-artifact@v7
         with:
           workflow: compat.yml
           workflow_conclusion: completed
@@ -130,7 +130,7 @@ jobs:
           branch: main
           path: dl
       - name: Download the log
-        uses: dawidd6/action-download-artifact@v6
+        uses: dawidd6/action-download-artifact@v7
         with:
           workflow: compat.yml
           workflow_conclusion: completed

--- a/.github/workflows/compat.yml
+++ b/.github/workflows/compat.yml
@@ -27,7 +27,7 @@ jobs:
         shell: bash
         run: |
           # Enable sources & install dependencies
-          sudo find /etc/apt/sources.list* -type f -exec sed -i 'p; s/^deb /deb-src /' '{}' +
+          sudo sed -i 's/^Types: deb$/Types: deb deb-src/' /etc/apt/sources.list.d/ubuntu.sources
           sudo apt-get update
           sudo apt-get build-dep findutils
       - name: Run GNU tests

--- a/.github/workflows/compat.yml
+++ b/.github/workflows/compat.yml
@@ -4,6 +4,9 @@ name: External-testsuites
 
 jobs:
   gnu-tests:
+    permissions:
+      actions: read
+
     name: Run GNU findutils tests
     runs-on: ubuntu-latest
     steps:
@@ -38,6 +41,7 @@ jobs:
       - name: Extract testing info
         shell: bash
         run: |
+
       - name: Upload gnu-test-report
         uses: actions/upload-artifact@v4
         with:
@@ -51,24 +55,43 @@ jobs:
         with:
           name: gnu-result
           path: gnu-result.json
-      - name: Download the result
-        uses: dawidd6/action-download-artifact@v7
+      - name: Download artifacts (gnu-result and gnu-test-report)
+        uses: actions/github-script@v7
         with:
-          workflow: compat.yml
-          workflow_conclusion: completed
-          name: gnu-result
-          repo: uutils/findutils
-          branch: main
-          path: dl
-      - name: Download the log
-        uses: dawidd6/action-download-artifact@v7
-        with:
-          workflow: compat.yml
-          workflow_conclusion: completed
-          name: gnu-test-report
-          repo: uutils/findutils
-          branch: main
-          path: dl
+          script: |
+            let fs = require('fs');
+            fs.mkdirSync('${{ github.workspace }}/dl', { recursive: true });
+
+            async function downloadArtifact(artifactName) {
+              // List all artifacts from the workflow run
+              let artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: ${{ github.run_id }},
+              });
+
+              // Find the specified artifact
+              let matchArtifact = artifacts.data.artifacts.find((artifact) => artifact.name === artifactName);
+              if (!matchArtifact) {
+                throw new Error(`Artifact "${artifactName}" not found.`);
+              }
+
+              // Download the artifact
+              let download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifact.id,
+                archive_format: 'zip',
+              });
+
+              // Save the artifact to a file
+              fs.writeFileSync(`${{ github.workspace }}/dl/${artifactName}.zip`, Buffer.from(download.data));
+            }
+
+            // Download the required artifacts
+            await downloadArtifact("gnu-result");
+            await downloadArtifact("gnu-test-report");
+
       - name: Compare failing tests against master
         shell: bash
         run: |
@@ -76,6 +99,8 @@ jobs:
       - name: Compare against main results
         shell: bash
         run: |
+          unzip dl/gnu-result.zip -d dl/
+          unzip dl/gnu-test-report.zip -d dl/
           mv dl/gnu-result.json latest-gnu-result.json
           python findutils/util/compare_gnu_result.py
 
@@ -120,24 +145,42 @@ jobs:
         with:
           name: bfs-result
           path: bfs-result.json
-      - name: Download the result
-        uses: dawidd6/action-download-artifact@v7
+      - name: Download artifacts (gnu-result and bfs-test-report)
+        uses: actions/github-script@v7
         with:
-          workflow: compat.yml
-          workflow_conclusion: completed
-          name: bfs-result
-          repo: uutils/findutils
-          branch: main
-          path: dl
-      - name: Download the log
-        uses: dawidd6/action-download-artifact@v7
-        with:
-          workflow: compat.yml
-          workflow_conclusion: completed
-          name: bfs-test-report
-          repo: uutils/findutils
-          branch: main
-          path: dl
+          script: |
+            let fs = require('fs');
+            fs.mkdirSync('${{ github.workspace }}/dl', { recursive: true });
+
+            async function downloadArtifact(artifactName) {
+              // List all artifacts from the workflow run
+              let artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: ${{ github.run_id }},
+              });
+
+              // Find the specified artifact
+              let matchArtifact = artifacts.data.artifacts.find((artifact) => artifact.name === artifactName);
+              if (!matchArtifact) {
+                throw new Error(`Artifact "${artifactName}" not found.`);
+              }
+
+              // Download the artifact
+              let download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifact.id,
+                archive_format: 'zip',
+              });
+
+              // Save the artifact to a file
+              fs.writeFileSync(`${{ github.workspace }}/dl/${artifactName}.zip`, Buffer.from(download.data));
+            }
+
+            // Download the required artifacts
+            await downloadArtifact("bfs-result");
+            await downloadArtifact("bfs-test-report");
       - name: Compare failing tests against main
         shell: bash
         run: |
@@ -145,6 +188,8 @@ jobs:
       - name: Compare against main results
         shell: bash
         run: |
+          unzip dl/bfs-result.zip -d dl/
+          unzip dl/bfs-test-report.zip -d dl/
           mv dl/bfs-result.json latest-bfs-result.json
           python findutils/util/compare_bfs_result.py
 

--- a/src/find/matchers/logical_matchers.rs
+++ b/src/find/matchers/logical_matchers.rs
@@ -356,7 +356,6 @@ impl Matcher for NotMatcher {
 }
 
 #[cfg(test)]
-
 mod tests {
     use super::*;
     use crate::find::matchers::quit::QuitMatcher;

--- a/src/find/matchers/mod.rs
+++ b/src/find/matchers/mod.rs
@@ -143,7 +143,7 @@ pub struct MatcherIO<'a> {
     deps: &'a dyn Dependencies,
 }
 
-impl<'a> MatcherIO<'a> {
+impl MatcherIO<'_> {
     pub fn new(deps: &dyn Dependencies) -> MatcherIO<'_> {
         MatcherIO {
             should_skip_dir: false,

--- a/src/find/matchers/prune.rs
+++ b/src/find/matchers/prune.rs
@@ -24,8 +24,8 @@ impl Matcher for PruneMatcher {
         true
     }
 }
-#[cfg(test)]
 
+#[cfg(test)]
 mod tests {
     use super::*;
     use crate::find::matchers::tests::get_dir_entry_for;

